provider/datagen/src/registry.rs

```diff
@@ -4,199 +4,189 @@
 
-/// List of all supported keys
-pub fn all_keys() -> Vec<DataKey> {
-    let mut v = vec![
-        icu_calendar::provider::JapaneseErasV1Marker::KEY,
-        icu_datetime::provider::calendar::DatePatternsV1Marker::KEY,
-        icu_datetime::provider::calendar::TimePatternsV1Marker::KEY,
-        icu_datetime::provider::calendar::DateSkeletonPatternsV1Marker::KEY,
-        icu_datetime::provider::calendar::DateSymbolsV1Marker::KEY,
-        icu_datetime::provider::calendar::TimeSymbolsV1Marker::KEY,
-        icu_datetime::provider::time_zones::TimeZoneFormatsV1Marker::KEY,
-        icu_datetime::provider::time_zones::ExemplarCitiesV1Marker::KEY,
-        icu_datetime::provider::time_zones::MetaZoneGenericNamesLongV1Marker::KEY,
-        icu_datetime::provider::time_zones::MetaZoneGenericNamesShortV1Marker::KEY,
-        icu_datetime::provider::time_zones::MetaZonePeriodV1Marker::KEY,
-        icu_datetime::provider::time_zones::MetaZoneSpecificNamesLongV1Marker::KEY,
-        icu_datetime::provider::time_zones::MetaZoneSpecificNamesShortV1Marker::KEY,
-        icu_datetime::provider::week_data::WeekDataV1Marker::KEY,
-        icu_decimal::provider::DecimalSymbolsV1Marker::KEY,
-        icu_list::provider::AndListV1Marker::KEY,
-        icu_list::provider::OrListV1Marker::KEY,
-        icu_list::provider::UnitListV1Marker::KEY,
-        icu_locale_canonicalizer::provider::AliasesV1Marker::KEY,
-        icu_locale_canonicalizer::provider::LikelySubtagsV1Marker::KEY,
-        icu_plurals::provider::CardinalV1Marker::KEY,
-        icu_plurals::provider::OrdinalV1Marker::KEY,
-        icu_provider_adapters::fallback::provider::LocaleFallbackLikelySubtagsV1Marker::KEY,
-        icu_provider_adapters::fallback::provider::LocaleFallbackParentsV1Marker::KEY,
-        #[cfg(feature = "experimental")]
-        icu_casemapping::provider::CaseMappingV1Marker::KEY,
-        icu_normalizer::provider::CanonicalDecompositionDataV1Marker::KEY,
-        icu_normalizer::provider::CompatibilityDecompositionSupplementV1Marker::KEY,
-        #[cfg(feature = "experimental")]
-        icu_normalizer::provider::Uts46DecompositionSupplementV1Marker::KEY,
-        icu_normalizer::provider::CanonicalDecompositionTablesV1Marker::KEY,
-        icu_normalizer::provider::CompatibilityDecompositionTablesV1Marker::KEY,
-        icu_normalizer::provider::CanonicalCompositionsV1Marker::KEY,
-        icu_normalizer::provider::CanonicalCompositionPassthroughV1Marker::KEY,
-        icu_normalizer::provider::CompatibilityCompositionPassthroughV1Marker::KEY,
-        #[cfg(feature = "experimental")]
-        icu_normalizer::provider::Uts46CompositionPassthroughV1Marker::KEY,
-        icu_normalizer::provider::NonRecursiveDecompositionSupplementV1Marker::KEY,
-    ];
-    v.extend(icu_properties::provider::ALL_KEYS);
-    #[cfg(feature = "experimental")]
-    v.extend(icu_segmenter::ALL_KEYS);
-    v.extend(crate::transform::icuexport::collator::ALL_KEYS);
```

Comment:
Discussion: I prefer referring to these external lists of keys rather than duplicating them here.

Reply:
We need a list of markers here, which cannot be assembled from the lists of keys we have in some places. And because the list of markers gives us the list of keys, there's no need to piece the keys together like we did before.

Reply:
hmm. ok. I guess one advantage of having the list here is that it may be easier to have a different set of unicode properties for testdata.

Reply:
The API of this doesn't change, so I don't understand how this would change anything for testdata?

We enumerated all these markers before, just split into separate `make_exportable_provider` calls in the transformers instead of in one place.